fs2-aws-s3/src/main/scala/fs2/aws/s3/S3.scala

```diff
@@ -106,20 +117,31 @@ object S3 {
             ).map(_.eTag() -> i.toInt)
           }
 
-        def completeUpload(uploadId: UploadId): Pipe[F, List[(PartETag, PartId)], ETag] =
+        def completeUpload(uploadId: UploadId): Pipe[F, List[(PartETag, PartId)], Option[ETag]] =
```

Comment:
Can we consider a NonEmptyList of parts to complete and make the user handle this situation?

Reply:
Is your suggestion is that the return type would look something like:
`Pipe[F, List[(PartETag, PartId)], Option[NonEmptyList[(PartETag, PartId)]]`?

In the case of no data/empty file. There will be no parts at all to "complete". I'm not sure if its good to expect the user to "complete" the upload either. This could be a bigger breaking change to someone upgrading to a newer version of this library, and isn't aware that their uploads will be lost of they don't "complete" them. Let me know what you think.